### PR TITLE
FileIO modularization: first part

### PR DESCRIPTION
**Pull request by TomFischer**
In order to remove dependencies the PR is the first step of modularization, see issue https://github.com/ufz/ogs/issues/1164. The PR is mainly moving files and change namespaces. There isn't any new line of code except the CMakeLists.txt's.

### CODE REVIEW

File: BaseLib/CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +26,16 @@ if(MSVC)
 	target_link_libraries(BaseLib INTERFACE WinMM) # needed for timeGetTime
 endif()
 
+if(QT4_FOUND)
+	target_link_libraries(BaseLib PUBLIC Qt4::QtXml Qt4::QtXmlPatterns)
+	if(WIN32 AND CMAKE_CROSSCOMPILING)
+		find_package(OpenSSL)
```
Comment (norihiro-w) [inline BaseLib/CMakeLists.txt]: Can you move this to Find.cmake?

Comment (bilke) [inline BaseLib/CMakeLists.txt]: Put this in `scripts/cmake/Find.cmake`:

```cmake
find_package(OpenSSL)
```

And this into the `CMakeLists.txt`-file:

```cmake
if(WIN32 AND CMAKE_CROSSCOMPILING AND OPENSSL_FOUND)
    target_link_libraries(BaseLib PUBLIC Qt4::QtNetwork ${OPENSSL_LIBRARIES} ws2_32)
endif()
```

**Comment by norihiro-w** [comment]
looks ok. please resolve conflicts

File: BaseLib/CMakeLists.txt [hunk 2]
```diff
@@ -15,6 +26,13 @@ if(MSVC)
 	target_link_libraries(BaseLib INTERFACE WinMM) # needed for timeGetTime
 endif()
 
+if(QT4_FOUND)
+	target_link_libraries(BaseLib PUBLIC Qt4::QtXml Qt4::QtXmlPatterns)
+	if(WIN32 AND CMAKE_CROSSCOMPILING AND OPENSSL_FOUND)
+		target_link_libraries(BaseLib PUBLIC Qt4::QtNetwork ${OPENSSL_LIBRARIES} ws2_32)
```
Comment (endJunction) [inline BaseLib/CMakeLists.txt]: The explicit `PUBLIC` and `INTERFACE` keywords in all of the `target_link_libraries` are removed now. The default behaviour (transitive) is sufficient, no need for finer grained dependencies for now.

**Comment by wenqing** [comment]
:+1:

**Comment by endJunction** [comment]
@TomFischer  there is a small addendum in your PRs

**Pull request merged**